tla-pp/src/test/scala/at/forsyte/apalache/tla/pp/TestKeramelizer.scala

```diff
@@ -291,4 +296,26 @@ class TestKeramelizer extends FunSuite with BeforeAndAfterEach {
     assert(expected == output)
   }
 
+  test("simplifies TLA+ expressions to KerA+") {
+    // Generator for PBT
+    val gens = new IrGenerators { override val maxArgs: Int = 3 }
+    // Generated operators
+    val ops = gens.simpleOperators ++ gens.logicOperators ++ gens.arithOperators ++ gens.setOperators
+    // Predicates for Keramelizer input and Keramelizer output (= KerA+)
+    val inputPred = KeramelizerInputLanguagePred()
+    val outputPred = KeraLanguagePred()
+
+    val prop = forAll(gens.genTlaEx(ops)(gens.emptyContext)) { ex =>
+      inputPred.isExprOk(ex) == PredResultOk() ==>
+        (try {
+          val keramelized = keramelizer(ex)
+          val inKera = outputPred.isExprOk(keramelized)
+          inKera shouldBe PredResultOk() withClue s"when keramelizing $ex to $keramelized"
+          true
+        } catch {
+          case _: MalformedTlaError => true
```

Comment:
It's worth noting that this I think this has the potential to make the test very ineffective, since its possible the rng could roll out 1000 malformed Tla expression, meaning this test would always pass but never once exercise the intended functionality!

Reply:
Oh! Perhaps I'm mistaken here. I see we have the conditional checking whether the `isExprOk` is true as an antecedent, and ScalaCheck will warn us if too few conditions satisfy the antecedent.

But why do we need the `catch` if we are first ensuring that `ex` is OK?

Reply:
On idea to move this into an antecedent check that will warn on too many discarded results: You might use https://www.scala-lang.org/api/2.12.4/scala/util/control/Exception$.html to reify the exception into an `Option`. You could then use either a `suchThat` generator to only take `Some` or just a boolean `isSome` check in the antecedent.

Reply:
Addressed in 41d4476 by adding checks to the language predicate that prevent throwing of the exception. I believe this is the best place for them.

@shonfeder PTAL?